README.md

```diff
@@ -1,32 +1,45 @@
 ![test](https://github.com/ryukinix/hazard-memory/workflows/test/badge.svg)
 
 # hazard-memory
-An experimental  SDL2 Platform Game written in C++
-
-# Roadmap
-## Game Engine
-
-- [ ] Setup basic scaffold.
-- [ ] Implementation of a ECS system.
-- [ ] Implementation of Hero, NPCs, Items and Enemies Entities.
-- [ ] Implementation of Hero, NPCs, Items and Enemies Behaviours.
-- [ ] Implementation of Animation system (Using Quads).
-- [ ] Implementation of GUI basics, made to be easily extensible (Animation is optional).
-- [ ] Implementation of Configuration files system load. (JSON, .INI or a created encrypted file extension)
-- [ ] Implementation of Configuration in-game setup (Resolution scaling, Input configuration: Joystick or Keyboard).
-- [ ] Implementation of Conversation load files and system of control and render flow. (Text could be animated).
-- [ ] Implementation of Tile system. (Including parsing and load files with levels data).
-- [ ] Implementation of Basic collision system.
-- [ ] Implementation of a Battle system.
-- [ ] Implementation of Items bag and usage.
-- [ ] Implementation of Gameplay Load/Save system. (Checkpoint or Save State system).
-
-## Game Design
-
-- [ ] Define a theme for the world.
-- [ ] Define a main plot.
-- [ ] Draw graphical user interface assets.
-- [ ] Draw assets for the hero, enemies(including bosses), and stages.
-- [ ] Draw assets for items and armor.
-- [ ] Draw assets for the world and dungeons.
-- [ ] Draw a game ending scene.
+An experimental game engine for 2D games based in SDL2 written in C++. 
+
+# Requirements
+
+- [SDL2]
+- [SDL2_gfx]
+- [g++]
+- [GNU Make]
+
+[SDL2]: https://www.libsdl.org/download-2.0.php
+[SDL2_gfx]: https://www.ferzkopp.net/Software/SDL2_gfx/Docs/html/index.html
+[g++]: https://gcc.gnu.org/
+[GNU Make]: https://www.gnu.org/software/make/manual/make.html 
+
+## How to install dependencies
+
+### Ubuntu
+
+```bash
+sudo apt install libsdl2-dev libsdl2-gfx-dev  gcc -y 
```

Comment:
Why libs here are dev version? This wasn't to be here, for dev purposes, use a topic with `Dev` label for this and provide commands to install the libs only for hazard internet development. And maintain the `normal` commands only for game development.

Reply:
dev version include the .h files, which is necessary to compile. This is not a branch or version. Packages in ubuntu not labelled as dev only includes the dynamic libraries.

The flag `-y` is not necessary, only install by default. I can remove it.

Reply:
flag removed, now the libs will be not installed until the user confirmation

Reply:
```
❯  dpkg -L libsdl2-dev
/.
/usr
/usr/bin
/usr/bin/sdl2-config
/usr/include
/usr/include/SDL2
/usr/include/SDL2/SDL.h
/usr/include/SDL2/SDL_assert.h
/usr/include/SDL2/SDL_atomic.h
/usr/include/SDL2/SDL_audio.h
/usr/include/SDL2/SDL_bits.h
/usr/include/SDL2/SDL_blendmode.h
/usr/include/SDL2/SDL_clipboard.h
/usr/include/SDL2/SDL_config.h
/usr/include/SDL2/SDL_cpuinfo.h
/usr/include/SDL2/SDL_egl.h
/usr/include/SDL2/SDL_endian.h
/usr/include/SDL2/SDL_error.h
/usr/include/SDL2/SDL_events.h
/usr/include/SDL2/SDL_filesystem.h
/usr/include/SDL2/SDL_gamecontroller.h
/usr/include/SDL2/SDL_gesture.h
/usr/include/SDL2/SDL_haptic.h
/usr/include/SDL2/SDL_hints.h
/usr/include/SDL2/SDL_joystick.h
/usr/include/SDL2/SDL_keyboard.h
/usr/include/SDL2/SDL_keycode.h
/usr/include/SDL2/SDL_loadso.h
/usr/include/SDL2/SDL_log.h
/usr/include/SDL2/SDL_main.h
/usr/include/SDL2/SDL_messagebox.h
/usr/include/SDL2/SDL_mouse.h
/usr/include/SDL2/SDL_mutex.h
/usr/include/SDL2/SDL_name.h
/usr/include/SDL2/SDL_opengl.h
/usr/include/SDL2/SDL_opengl_glext.h
/usr/include/SDL2/SDL_opengles.h
/usr/include/SDL2/SDL_opengles2.h
/usr/include/SDL2/SDL_opengles2_gl2.h
/usr/include/SDL2/SDL_opengles2_gl2ext.h
/usr/include/SDL2/SDL_opengles2_gl2platform.h
/usr/include/SDL2/SDL_opengles2_khrplatform.h
/usr/include/SDL2/SDL_pixels.h
/usr/include/SDL2/SDL_platform.h
/usr/include/SDL2/SDL_power.h
/usr/include/SDL2/SDL_quit.h
/usr/include/SDL2/SDL_rect.h
/usr/include/SDL2/SDL_render.h
/usr/include/SDL2/SDL_revision.h
/usr/include/SDL2/SDL_rwops.h
/usr/include/SDL2/SDL_scancode.h
/usr/include/SDL2/SDL_sensor.h
/usr/include/SDL2/SDL_shape.h
/usr/include/SDL2/SDL_stdinc.h
/usr/include/SDL2/SDL_surface.h
/usr/include/SDL2/SDL_system.h
/usr/include/SDL2/SDL_syswm.h
/usr/include/SDL2/SDL_test.h
/usr/include/SDL2/SDL_test_assert.h
/usr/include/SDL2/SDL_test_common.h
/usr/include/SDL2/SDL_test_compare.h
/usr/include/SDL2/SDL_test_crc32.h
/usr/include/SDL2/SDL_test_font.h
/usr/include/SDL2/SDL_test_fuzzer.h
/usr/include/SDL2/SDL_test_harness.h
/usr/include/SDL2/SDL_test_images.h
/usr/include/SDL2/SDL_test_log.h
/usr/include/SDL2/SDL_test_md5.h
/usr/include/SDL2/SDL_test_memory.h
/usr/include/SDL2/SDL_test_random.h
/usr/include/SDL2/SDL_thread.h
/usr/include/SDL2/SDL_timer.h
/usr/include/SDL2/SDL_touch.h
/usr/include/SDL2/SDL_types.h
/usr/include/SDL2/SDL_version.h
/usr/include/SDL2/SDL_video.h
/usr/include/SDL2/SDL_vulkan.h
/usr/include/SDL2/begin_code.h
/usr/include/SDL2/close_code.h
/usr/include/x86_64-linux-gnu
/usr/include/x86_64-linux-gnu/SDL2
/usr/include/x86_64-linux-gnu/SDL2/_real_SDL_config.h
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/cmake
/usr/lib/x86_64-linux-gnu/cmake/SDL2
/usr/lib/x86_64-linux-gnu/cmake/SDL2/sdl2-config.cmake
/usr/lib/x86_64-linux-gnu/libSDL2.a
/usr/lib/x86_64-linux-gnu/libSDL2_test.a
/usr/lib/x86_64-linux-gnu/libSDL2main.a
/usr/lib/x86_64-linux-gnu/pkgconfig
/usr/lib/x86_64-linux-gnu/pkgconfig/sdl2.pc
/usr/share
/usr/share/aclocal
/usr/share/aclocal/sdl2.m4
/usr/share/doc
/usr/share/doc/libsdl2-dev
/usr/share/doc/libsdl2-dev/copyright
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/sdl2-config.1.gz
/usr/include/x86_64-linux-gnu/SDL2/SDL_platform.h
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so
/usr/lib/x86_64-linux-gnu/libSDL2.so
/usr/share/doc/libsdl2-dev/changelog.Debian.gz
```

```
❯ dpkg -L libsdl2-2.0-0:amd64 
/.
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0.10.0
/usr/share
/usr/share/doc
/usr/share/doc/libsdl2-2.0-0
/usr/share/doc/libsdl2-2.0-0/BUGS.txt
/usr/share/doc/libsdl2-2.0-0/CREDITS.txt
/usr/share/doc/libsdl2-2.0-0/README-SDL.txt
/usr/share/doc/libsdl2-2.0-0/README.txt
/usr/share/doc/libsdl2-2.0-0/changelog.Debian.gz
/usr/share/doc/libsdl2-2.0-0/copyright
/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0
```

Reply:
Ok, fine. Could i merge?